src/source/CppGenerator.scala

```diff
@@ -297,6 +300,16 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
             w.wl(s"virtual $ret ${idCpp.method(m.ident)}${params.mkString("(", ", ", ")")}$constFlag = 0;")
           }
         }
+        // Properties
```

Comment:
It would be nice if this shared more code with the method-generating codepaths, since that will help to keep the code simple and make it likely to all be updated in sync with future changes.  You should be able to create method definitions from the properties, then use the normal method-generation code.

One option to consider (though it might be too aggressive) would be to implement property generation in most languages as a transformation on the AST, by generating a new AST with methods in place of properties, then running the normal generator on them.  That could be a suitable approach for all generators which don't have special support for properties (i.e. ObjC with its @property declaration).